OSDOCS-1770 Installing a cluster on GCP in a restricted network (#32482)
openshift/openshift-docs: installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc

[id="installing-restricted-networks-gcp-installer-provisioned"]
= Installing a cluster on GCP in a restricted network
include::modules/common-attributes.adoc[]
:context: installing-restricted-networks-gcp-installer-provisioned

toc::[]

In {product-title} version {product-version}, you can install a cluster on Google Cloud Platform (GCP) in a restricted network by creating an internal mirror of the installation release content on an existing Google Virtual Private Cloud (VPC).

[IMPORTANT]
====

> While you can install an {product-title} cluster by using mirrored installation release content, your cluster still requires internet access to use the GCP APIs.

Reviewer: ```suggestion
You can install an {product-title} cluster by using mirrored installation release content, but your cluster will require internet access to use the GCP APIs.
```

====

[id="prerequisites_installing-restricted-networks-gcp-installer-provisioned"]
== Prerequisites

* You xref:../../installing/install_config/installing-restricted-networks-preparations.adoc#installing-restricted-networks-preparations[created a mirror registry on your bastion host] and obtained the `imageContentSources` data for your version of {product-title}.
+
[IMPORTANT]
====
Because the installation media is on the bastion host, use that computer to complete all installation steps.
====

> * You have an existing VPC in GCP. When installing to a restricted network using installer-provisioned infrastructure, you cannot use the installer-provisioned VPC. You must use a user-provisioned VPC that satisfies one of the following requirements:

Reviewer: ```suggestion
* You have an existing VPC in GCP. While installing a cluster in a restricted network that uses installer-provisioned infrastructure, you cannot use the installer-provisioned VPC. You must use a user-provisioned VPC that satisfies one of the following requirements:
```

Not sure on "While," specifically. Could also be "If."

> ** Contains the mirror registry.

Reviewer: ISG: Don't punctuate the ends of fragments.

Introductory sentence into the list reads a little awk to me. Not major, but maybe consider rephrasing.

** Has firewall rules or a peering connection to access the mirror registry hosted elsewhere.
* You reviewed details about the xref:../../architecture/architecture-installation.adoc#architecture-installation[{product-title} installation and update] processes.

> * If you use a firewall, you must xref:../../installing/install_config/configuring-firewall.adoc#configuring-firewall[configure it to allow the sites] that your cluster requires access to. While you might need to grant access to more sites, you must grant access to `*.googleapis.com` and `accounts.google.com`.

Reviewer: Consider rephrasing or splitting.

> * If you do not allow the system to manage identity and access management (IAM), then a cluster administrator can xref:../../installing/installing_gcp/manually-creating-iam-gcp.adoc#manually-creating-iam-gcp[manually create and maintain IAM credentials]. Manual mode can also be used in environments where the cloud IAM APIs are not reachable.

Reviewer: Must the cluster admin? Not sure if this is a prereq or an option.

Author: Yeah, I think this is a tweener that gives suggestions for a particular scenario. Since this is reproduced throughout the AWS/Azure/GCP install assemblies, I'll leave it for now and follow up with a possible stronger statement.
